### PR TITLE
WIP Instrumenting Failed and Errored jobs

### PR DESCRIPTION
**Pull request by zmoazeni**
This is mega-rough, but submitting a WIP draft to iterate on.

This is currently breaking the test suite. It also doesn't have any automated tests. May need to restructure the code in order to allow this to be easily tested in a handler-agnostic way.

### CODE REVIEW

File: lib/worker/index.js [hunk 1]
```diff
@@ -39,13 +41,15 @@ Object.keys(queues).forEach(name => {
...
     Sentry.captureException(err)
+    statsd.increment('job-error')
   })
 
   queue.on('failed', async (job, err) => {
     app.log.error(`Error occurred while processing job id=${job.id} on queue name=${name}`)
 
     const subscription = await Subscription.getSingleInstallation(job.data.jiraHost, job.data.installationId)
     await subscription.update({ syncStatus: 'FAILED' })
+    statsd.increment('job-failed')
```
Comment (zmoazeni) [inline lib/worker/index.js]: I'm also wondering if there are any appropriate tags to pass across. But I'm not sure yet if there are.

File: lib/worker/index.js [hunk 2]
```diff
@@ -39,13 +41,15 @@ Object.keys(queues).forEach(name => {
 
     Sentry.setTag('queue', name)
     Sentry.captureException(err)
+    statsd.increment('job-error')
```
Comment (jnraine) [inline lib/worker/index.js]: Metric names is always a bit up in the air but for the first few in this app, I've followed this pattern: `<prefix>.<area/concern>`. This has led to metric names like:

- `jira_integration.github_requests.95th`
- `jira_integration.jira_requests.95th`

In this same vein, what do you think about creating a `jobs` namespace and adding states beneath that?

- `jira_integration.jobs.started`
- `jira_integration.jobs.completed`
- `jira_integration.jobs.errored`
- etc

**Comment by zmoazeni** [comment]
> One stretch goal would be to use histogram instead of increment. In addition to counts, this would capture timing data for each job.

Yeah totally. I was planning on doing that in a separate effort. One metric for histogram/timing information for all jobs. A different metric for the number of errors and failures we encounter.

But I have no problem storing the number of failures/errors as a histogram. That'll give us both timing for just failures/errors as well as counts.

**Comment by jnraine** [comment]
>  I was planning on doing that in a separate effort.

👍 Shipping the counts now and the histograms later sounds good.

My one recommendation would be to use the same count key as the histogram:

```javascript
statsd.increment('jira_integration.jobs.started.count')
# then later
statsd.histogram('jira_integration.jobs.started')
```

While this isn't required, it ensures the metric name doesn't change when the histogram work is done.

You may have already considered recommendation. If so, carry on! 😄

**Comment by zmoazeni** [comment]
> You may have already considered recommendation.

Oh I didn't! Good idea!

**Comment by zmoazeni** [comment]
This is all set as of https://github.com/integrations/jira/pull/312

**Pull request closed without merging**